pkg/proxy/audit/audit.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package audit
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	genericapifilters "k8s.io/apiserver/pkg/endpoints/filters"
+	"k8s.io/apiserver/pkg/server"
+	genericfilters "k8s.io/apiserver/pkg/server/filters"
+
+	"github.com/jetstack/kube-oidc-proxy/cmd/app/options"
+)
+
+type Audit struct {
```

Comment:
Some comments on this structure/the methods it defines would help to understand it I think. Especially on the `With*` methods... they confuse me a bit (it looks like it is injecting an audit function into the handler chain?  But I don't know what a LongRunningFunc is for, and `genericapifilters.WithRequestInfo(handler, a.serverConfig.RequestInfoResolver)` I'm also a bit uncertain on.

Is the `handler` arg the 'next' handler in the chain?